### PR TITLE
Respect Descriptor.payload & Descriptor.xmlNamespace

### PR DESCRIPTION
**Pull request by xvrh**
It fixes 20 tests (mainly in rest-xml testsuite).

### CODE REVIEW

File: generated/aws_cloudfront_api/lib/cloudfront-2016-11-25.dart [hunk 1]
```diff
@@ -4773,18 +4839,21 @@ class Origin {
...
-      customOriginConfig.toXml('CustomOriginConfig'),
+      _s.encodeXmlStringValue('DomainName', domainName),
       _s.encodeXmlStringValue('OriginPath', originPath),
-      s3OriginConfig.toXml('S3OriginConfig'),
+      customHeaders?.toXml('CustomHeaders'),
+      s3OriginConfig?.toXml('S3OriginConfig'),
+      customOriginConfig?.toXml('CustomOriginConfig'),
+    ];
+    final $attributes = <_s.XmlAttribute>[
+      ...?attributes,
```
Comment (Schwusch) [inline generated/aws_cloudfront_api/lib/cloudfront-2016-11-25.dart:4852]: wow, I didn't know you could do null safe destructuring :O

**Review by Schwusch — APPROVED** [review]

**Comment by Schwusch** [comment]
wow, given how much you've fixed I'm surprised anything has ever worked

File: generated/aws_cloudfront_api/lib/cloudfront-2016-11-25.dart (hunk 1 above)
Comment (xvrh) [reply generated/aws_cloudfront_api/lib/cloudfront-2016-11-25.dart:4852]: Yes pretty cool. Called "null-aware spread operator"

**Comment by xvrh** [comment]
@Schwusch it has a big impacts on the generated tests but not that much on the actual APIs (relative to the number of API they are).

**Pull request merged**